### PR TITLE
Support Symfony 7

### PR DESCRIPTION
**Pull request by RobinDev**
PHPUnit is not setted in dev requirements. How are you testing the project ?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
-            "homepage": "https://github.com/knplabs/KnpGaufretteBundle/contributors"
-        }
-    ],
-    "require":      {
-        "php": ">=7.1",
-        "knplabs/gaufrette": "^0.11",
-        "symfony/config": "^4.4|^5.0|^6.0",
-        "symfony/dependency-injection": "^4.4|^5.0|^6.0",
-        "symfony/http-kernel": "^4.4|^5.0|^6.0"
+  "name": "knplabs/knp-gaufrette-bundle",
```
Comment (garak) [inline composer.json:2]: all these changes in spaces need to be reverted

**Review by garak — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
-        }
+    {
+      "name": "The contributors",
+      "homepage": "https://github.com/knplabs/KnpGaufretteBundle/contributors"
+    }
+  ],
+  "require": {
+    "php": ">=7.1",
+    "knplabs/gaufrette": "^0.11",
+    "symfony/config": "^4.4|^5.0|^6.0|^7.0",
```
Comment (garak) [inline composer.json:26]: Symfony 4 should be removed: supporting three major versions of Symfony is enough

Comment (RobinDev) [reply composer.json]: Ok, SF5 requires PHP 7.2.5 so i updated php requirement too.

File: composer.json (hunk 1 above)
Comment (garak) [reply composer.json]: That's irrelevant, this PR is not about changing the format of the file.
I understand indeed that the format is probably not the most standard possible, but it should be changed in a different PR.

**Comment by garak** [comment]
By the way, Gaufrette dropped support for PHP 7.2 and 7.3 already, let's align with that.

File: composer.json [hunk 3]
```diff
@@ -16,17 +16,17 @@
...
+        "symfony/dependency-injection": "^5.0|^6.0|^7.0",
+        "symfony/http-kernel": "^5.0|^6.0|^7.0"
     },
-    "require-dev": {
-        "symfony/phpunit-bridge": "^5.4",
-        "symfony/console": "^4.4|^5.0|^6.0",
-        "symfony/filesystem": "^4.0|^5.0|^6.0",
-        "symfony/yaml": "^4.0|^5.0|^6.0"
+    "require-dev":  {
+        "symfony/phpunit-bridge": "^5.4|^7.0",
```
Comment (garak) [inline composer.json:26]: let's drop version 5 for the bridge, the latest version is meant to be used with all Symfony versions

**Review by garak — CHANGES_REQUESTED** [review]

**Review by garak — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

**Comment by maxhelias** [comment]
I missed, you can remove some compatibility check like : https://github.com/KnpLabs/KnpGaufretteBundle/blob/master/DependencyInjection/FactoryConfiguration.php#L20-L26 or https://github.com/KnpLabs/KnpGaufretteBundle/blob/master/DependencyInjection/MainConfiguration.php#L36-L41

**Comment by 1ed** [comment]
@Nek- if you have a little time, could you please have a look or how could we move forward here? Thanks!

**Comment by Nek-** [comment]
Hello, I have no right on this repository anymore. Sorry I can't help. Please reach KNPLabs.

**Comment by garak** [comment]
Maybe @KevinArtus? Just a wild guess by looking at the assignees

**Review by alexpozzi — APPROVED** [review]

**Review by alexpozzi — CHANGES_REQUESTED** [review]
@RobinDev do you mind adding PHP 8.2 in the test matrix?

Right now it does not test SF 7 because of this reason (test should be refactored in a further PR to be sure to test it with the right SF versions).

**Comment by alexpozzi** [comment]
Once done I will merge and create a release :wink:

**Review by alexpozzi — APPROVED** [review]
Thank you!

**Pull request merged**